bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -875,3 +916,11 @@ func (r *KubeadmConfigReconciler) storeBootstrapData(ctx context.Context, scope
 	conditions.MarkTrue(scope.Config, bootstrapv1.DataSecretAvailableCondition)
 	return nil
 }
+
+func bootstrapDataFormat(spec bootstrapv1.KubeadmConfigSpec) bootstrapv1.Format {
+	if spec.IgnitionConfig != nil {
+		return bootstrapv1.Ignition
+	}
+
+	return spec.Format
+}
```

Comment:
Why not use `spec.Format` for everything here?

Reply:
This was to avoid specifying 2 fields to achieve single thing. If `spec.IgnitionConfig` is specified, it implies that `ignition` config is desired. But as discussed in other comment, I'l make direct use of `spec.Format` :+1:

Reply:
We use `spec.Format` now.